### ISSUE TITLE
For doc ranking task, which loss should I use?

### ISSUE THREAD

**Issue by guotong1988** [UKPLab/sentence-transformers#5]
![image](https://user-images.githubusercontent.com/4702353/62436391-1354f380-b772-11e9-9ab6-4eb30ad97e9a.png)

Given a query, rank the doc.

I guess MULTIPLE_NEGATIVES_RANKING_LOSS.

**Comment by nreimers** [comment]
It depends how your training data looks like.

MULTI_NEGATIVES_RANKING_LOSS assumes that the batch looks like this:
x_1, y_1
x_2, y_2
....
x_n, y_n

(x_i, y_i) is treated as a positive example, for example given query x_i and the correct answer y_i, all other possible pairs (i.e. x_i, y_j  for i != j) is treated as negative examples, i.e., y_j would be the wrong answer for query x_i.

**Comment by guotong1988** [comment]
Thank you! Are there any other loss function for doc ranking problem?

**Comment by nreimers** [comment]
Doc ranking is currently not the core of this framework. You could use triplet loss, with providing a positive example and a negative example (positive example should rank higher  than negative example).

Doc ranking is not my expertise, so I sadly don't know if there are any other good / well established doc ranking losses that are useful.

**Comment by guotong1988** [comment]
Thank you!
